components/Login/LoginModal.js

```diff
@@ -93,6 +144,124 @@ class LoginModal extends React.Component {
         });
     }
 
+    socialLogin(e, provider) {
+        e.preventDefault();
+        console.log('Hit on social login icon', provider);
+        this.provider = provider;
+
+        $('.ui.login.modal').modal('toggle');
+
+        //prepare localStorage
+        localStorage.setItem(MODI, 'login');
+        localStorage.setItem(NAME, '');
+
+        //observe storage
+        $(window).off('storage').on('storage', this.handleStorageEvent.bind(this));
+
+        //show hint before open tab
+        const Provider = this.getProviderName();
+        swal({
+            title: 'Information',
+            text: 'A new window of your browser will be opened where you could do a sign in on ' + Provider + '. Please do so.',
+            type: 'info',
+            confirmButtonText: 'Confirm',
+            confirmButtonClass: 'positive ui button',
+            buttonsStyling: false,
+            showCloseButton: false,
+            showCancelButton: false,
+            allowEscapeKey: false
+        })
+        .then(() => {
+            //create new tab
+            let url = 'http://authorizationservice.manfredfris.ch:3000/connect/' + provider;
+
+            let width = screen.width*0.75, height = screen.height*0.75;
+            if (width < 600)
+                width = screen.width;
+            if (height < 500)
+                height = screen.height;
+            let left = screen.width/2-width/2, topSpace = screen.height/2-height/2;
+
+            let win = window.open(url, '_blank', 'width='+width+',height='+height+',left='+left+',top='+topSpace+',toolbar=No,location=No,scrollbars=no,status=No,resizable=no,fullscreen=No');
+            win.focus();
+
+            return true;
+        })
+        .catch();
+    }
+
+    clickedFacebook(e) {
```

Comment:
Use a parameter to distinguish between different providers instead of implementing three functions with the (nearly) exact same content.